src/syslogd.c

```diff
@@ -3271,6 +3301,58 @@ static void logit(char *fmt, ...)
 	fflush(stdout);
 }
 
+static void notifier_add(struct notifiers *newn, const char *program)
+{
+	while (*program && isspace(*program))
+		program++;
+
+	/* Check whether it is accessible, regardless of TOCTOU */
+	if (!access(program, X_OK)) {
+		struct notifier *np;
+		size_t len;
+
+		len = strlen(program);
+
+		np = calloc(1, sizeof(*np) + len +1);
+		if (np) {
+			/* xxx Actually wastes space -- vararray? */
+			np->n_program = (char*)&np[1];
+			memcpy(np->n_program, program, len);
+			SIMPLEQ_INSERT_TAIL(newn, np, n_link);
+		} else
+			ERR("Cannot allocate memory for a notify program");
+	} else
+		logit("notify: non-existing, or not executable program\n");
+}
+
+static void notifier_invoke(const char *logfile)
+{
+	char *argv[3];
+	struct notifier *np;
+
+	logit("notify: rotated %s, invoking hooks\n", logfile);
+
+	SIMPLEQ_FOREACH(np, &nothead, n_link) {
+		switch (fork()) {
```

Comment:
Please save and, log PID, at least for debug purposes.

Reply:
ok, doing this via "childpid" as elsewhere in this file, and logging in same style, too.